### ISSUE TITLE
Extract text from individual pages (`recognizePage` using Node)

### ISSUE THREAD

**Issue by bennyzen** [scribeocr/scribe.js#19]
Just tried to use the `recognizePage` method under Node, but I get an error like `worker not initialized`. The `recognize` method takes forever on a large PDF, but works as it should. `recognizePage` just sits there and does nothing.

This is the error that is thrown:
```
Error: Worker not initialized
    at recognizeAndConvert2 (file:///home/ben/repos/xxx/node_modules/.pnpm/scribe.js-ocr@0.4.0/node_modules/scribe.js-ocr/js/worker/generalWorker.js:247:55)
    at MessagePort.handleMessage (file:///home/ben/repos/xxx/node_modules/.pnpm/scribe.js-ocr@0.4.0/node_modules/scribe.js-ocr/js/worker/generalWorker.js:383:5)
```

I need to extract single pages to add some additional meta-data to the extracted text that then goes into my storage.

What am I missing here to be able to recognize single pages using Node (CLI)? Maybe someone can give me a hint. I think this is a bug.

**Comment by Balearica** [comment]
Thanks for reporting.  The short answer is that there is currently no easy way to recognize only a single page in a PDF.  However, this is probably not something that would be difficult to add.  Out of curiosity, can you describe the use-case in more detail?  Is the idea that you're trying to fill title/author/etc. from a PDF that contains an entire book? 

Regarding the `recognizePage` function, the documentation appears to be misleading, and that should definitely be fixed.  The `recognizePage` function is a lower-level function that is used internally by `recognize`, and requires additional code to work correctly.  I did not intend for `recognizePage` to be advertised, but it looks like it was included in the auto-generated docs, and the description is misleading.  I will edit or remove to avoid confusion going forward.

**Comment by bennyzen** [comment]
First of all, thank you for your answer and your great lib. And yes, I was really hoping to be able to use that method to extract single pages. Would have made my life so much easier.

Regarding the actual use-case:
I'm building some kind of [RAG](https://en.wikipedia.org/wiki/Retrieval-augmented_generation) system with a lot of lengthy PDFs. While ingesting the extracted text from the pages (previously split into chunks) into the vector database, I would have added some meta data like page number, offset, etc. By doing so, the users would get a rephrased answer from the LLM and then some kind of link to the reference. Imagine you have huge PDFs with a lot of pages, it's not really helpful to say "it's somewhere in there, but I can't tell you where. You have to find it yourself".

Unfortunately, I'm not familiar with the internals of your lib, else I would have tried to contribute this functionality. As a workaround (I'm not really happy with it, as it's not really elegant), I now do images of the single pages with muPDF, use OCR to grab the text, and then ingest the single pages into my storage. I know, it's a huge detour.  

And hey, please keep up your good work! If you can somehow get it working, I would love to use it.

**Comment by bennyzen** [comment]
Oh wow, this is exactly what I need. Thank you so much.

**Comment by Balearica** [comment]
> Maybe this should be made clearer in the docs, as I'm surely not the only one looking for this kind of functionality nowadays.

Closing the issue as completed.  I've updated function/argument names as well as docs to be clearer per the discussion above.   `recognizePage` has been renamed to `recognizePageImp`, the JSDoc description was updated to clearly state it's a lower-level function for advanced users only, and it has been removed from the [list of documented API functions](https://github.com/scribeocr/scribe.js/blob/master/docs/API.md).  Additionally, the `minValue`/`maxValue` arguments to `exportData` and `download` have been renamed to `minPage`/`maxPage` and documented more clearly.

Agree it would be nice to have more detailed documentation for this use-case, however unfortunately there are so many outstanding development tasks that I personally don't have much time to devote to documentation in the short run.  If anybody else writes an article/vignette on Scribe.js for LLM use-cases I would merge it in and/or link to it.
